modules/tinymce/src/core/main/ts/util/Quirks.ts

```diff
@@ -678,6 +678,14 @@ const Quirks = (editor: Editor): Quirks => {
     }
   };
 
+  const dropDragEndEvent = () => {
+    editor.on('drop', (event) => {
+      if (event.dataTransfer?.getData('text/html').startsWith('<img')) {
```

Comment:
You only want to match on image elements not img and other content?

```suggestion
      if (/^<img[^>]*>$/.test(event.dataTransfer?.getData('text/html'))) {
```